### PR TITLE
Show mapbrowse

### PR DESCRIPTION
**Pull request by kencho51**
# Pull request for issue: #939

This is a pull request for the following functionalities:

<Insert description of the proposed new functionalities and how they can be tested>

Mapbrowse is not working in staging becase script src `https://openlayers.org/en/v4.6.5/build/ol.js` and image src from `openstreetmap.org` are blocked by the CSP.

By allowing the script-src and img-src in CSP, the mapbrowse could not work in staging as expected.

To make mapbrowse work in dev locally, geographical locations of Adelie penguin and Foxtail millet has been added into `data/dev/sample_attribute.csv`, which were found after some google searches: 

| Species | location | latitude and longitude |
| -------- | -------- | --------------------- |
|  Adelie penguin | south orkney islands, Antartica | -60.345999, -45.295999 |
|  Foxtail millet |  Hehei, China | 38.0360, 114.4698 |

This is a [coordinate converter](https://coordinates-converter.com/en/decimal/-60.637103,-45.241934?karte=OpenStreetMap&zoom=1).

Here is the screenshot of working mapbrowse in my dev:
image

Here is the screenshot of working mapbrowse in my staging:
image

### CODE REVIEW

**Review by pli888 — COMMENTED** [review]
The code looks good to me. It is worth thinking about adding a new dataset into `dev` data that contains a few sample data points, otherwise we always have to switch to another database to see the Map Browse functionality working on our local environments.

**Review by pli888 — APPROVED** [review]
I can see the Penguin and Foxtail Millet samples on the MapBrowse page. All of the tests pass on my CI pipeline and I can also see the samples on the MapBrowse page with production data on my staging server. I approve this PR.

**Comment by pli888** [comment]
Coming back to this PR, I was wondering how to write a test for the presence of the sample map on this page. Before the CSP was updated, there was just whitespace instead of the map.

**Review by rija — COMMENTED** [review]
Hi @kencho51,

All tests are passing locally and on CI.
And the map browse functionality functions nicely on my local environment and also on my staging environment.

The changes made to make the functionality work make sense to me, 

Augmenting the test data with geolocation values is a good idea. 

I have a question and a concern though:

Firstly, how do people know about this functionality, I couldn't find anywhere on the web site or the codebase a link to the mapbrowse functionality?

Secondly, as far as I know, we don't have any ways to know when that functionality breaks, and in addition to the CSP settings to be aware of, there is a non-trivial amount of code (mostly javascript, mostly legacy) to make it work.

I think this bug fix is an opportunity to start adding some test coverage, even a very basic acceptance/functional test, e.g that checks that the source of the page contains the locations that  were added to ``sample_attribute.csv`` is good enough to start the ball of gradual refinement rolling.

**Comment by rija** [comment]
> Coming back to this PR, I was wondering how to write a test for the presence of the sample map on this page. Before the CSP was updated, there was just whitespace instead of the map.

Hi @pli888, good thinking. 
My review comment above has a suggestion on how to do that (it pays to look a the source of a page)

**Comment by rija** [comment]
Hi @kencho51,  

i think it's better to use the data from the spreadsheet instead of hard coding them in the test. otherwise when we need to change the test data, we need to remember to also change the test.

 you can use something like this to read test data from PHP:
```

$testFixture = array_map('str_getcsv', file('sample_attribute.csv')); 

```

File: tests/functional/MapBrowseCest.php [hunk 1]
```diff
@@ -9,11 +9,19 @@ public function _before(FunctionalTester $I)
...
+
         $I->amOnPage("/site/mapbrowse");
         $I->seeResponseCodeIs(200);
-        $html =$I->grabPageSource();
-        $I->assertContains('"coordinates":[-45.295999 ,-60.345999]', $html, "Coordinate not found!");
-        $I->assertContains('"coordinates":[114.4698 ,38.0360]', $html, "Coordinate not found!");
-
+        $pageSource = $I->grabPageSource();
+        $I->assertContains($expectPenguinCoordinateXY, $pageSource, "Adelie penguin's coordinates not found!");
+        $I->assertContains($expectFoxtailMilletCoordinateXY, $pageSource, "Foxtail Millet's coordinates not found!");
```
Comment (rija) [inline tests/functional/MapBrowseCest.php:25]: Hi @kencho51,
those three lines (23,24, 25) can be replaced with the more concise:

```
       $I->canSeeInSource($expectPenguinCoordinateXY);
       $I->canSeeInSource($expectFoxtailMilletCoordinateXY);
```

Also the HTTP response test at line 22 doesn't bring any value as if there is a non 200 response, the tests below will fail anyways and Codeception will generate snapshot that can show you what the error was. So it's better deleting that check.

**Comment by pli888** [comment]
>I think a functional test is more appropriate with that example, as we are checking the integrity of the "data feed" rather than replicating the user's experience.

I think the `tryToSeeCoordinatesInPageSource` functional test works fine.

>Firstly, how do people know about this functionality, I couldn't find anywhere on the web site or the codebase a link to the mapbrowse functionality?

@rija A user needs to log into the GigaDB website and then the home page will display a `Browse Samples` link next to the `LogOut` link in the top navigation bar.

**Review by rija — APPROVED** [review]
HI @kencho51 
the code looks good now.
all tests are passing locally and on CI and the functionally works locally and on CI.
Therefore I'm happy to approve this PR.

**Pull request merged**